ingestion_server/test/unit_tests/test_distributed_reindex_scheduler.py

```diff
@@ -0,0 +1,87 @@
+import json
+from unittest import mock
+
+import pook
+import pytest
+
+from ingestion_server import distributed_reindex_scheduler
+
+
+@pytest.mark.parametrize(
+    "estimated_records, record_limit, workers, expected_ranges",
+    [
+        # One worker
+        (100, 1000, ["worker1"], [(0, 100)]),
+        # Multiple workers, even split
+        (100, 1000, ["worker1", "worker2"], [(0, 50), (50, 100)]),
+        # Multiple workers, uneven split
+        # NOTE: records get dropped in this case
+        (100, 1000, ["worker1", "worker2", "worker3"], [(0, 33), (33, 66), (66, 99)]),
+        # One worker, limited
+        (100, 55, ["worker1"], [(0, 55)]),
+        # Two workers, limited
+        (100, 50, ["worker1", "worker2"], [(0, 25), (25, 50)]),
+    ],
+)
+def test_assign_work(estimated_records, record_limit, workers, expected_ranges):
+    # Checks for the parameters
+    assert len(workers) == len(
+        expected_ranges
+    ), "Number of workers and expected ranges do not match, correct the test parameters"
+    # Set up database mock response
+    mock_db = mock.MagicMock()
+    mock_db.cursor.return_value.__enter__.return_value.fetchone.return_value = [
+        estimated_records
+    ]
+    # Enable pook & mock other internal functions
+    with pook.use(), mock.patch(
+        "ingestion_server.distributed_reindex_scheduler.get_record_limit"
+    ) as mock_get_record_limit, mock.patch(
+        "ingestion_server.distributed_reindex_scheduler._wait_for_healthcheck"
+    ) as mock_wait_for_healthcheck:
+        mock_wait_for_healthcheck.return_value = True
+        mock_get_record_limit.return_value = record_limit
+
+        post_mocks = [
+            pook.post(f"http://{worker}:8002/indexing_task") for worker in workers
+        ]
+
+        distributed_reindex_scheduler._assign_work(
+            mock_db,
+            workers,
+            "sample_model",
+            "sample_table",
+            "sample_index",
+        )
+
+        for worker, mock_post, (start_id, end_id) in zip(
+            workers, post_mocks, expected_ranges
+        ):
+            assert mock_post.calls > 0
```

Comment:
Just a suggestion to use `mock_post.matched` (which itself does `.calls > 0`), or probably better, `mock_post.done` to ensure it was matched the expected number of times (in this case, once). `matched` and `calls > 0` just checks if it was matched _at all_, not that it was matched the expected number of times. In this case it is 1, so there's no _real_ difference, but semantically, and when considering the approach to take for other tests that may have multi-use mocks, `.done` is probably a more consistent and reliable assertion.